### PR TITLE
Add mcap local file data source

### PR DESCRIPTION
**Pull request by defunctzombie**
**User-Facing Changes**
None.

**Description**
The mcap data source adds support for loading mcap files. The data source requires the "mcap data source" feature flag to show up in the data sources list.
    
The mcap data source is highly experimental with the file format still evolving. This change supports the "ros1", "ros2", and "protobuf" schema types.

### CODE REVIEW

File: packages/studio-base/src/randomAccessDataProviders/McapDataProvider.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+          if (record.schema === "ros1") {
+            const parsedDefinitions = parseMessageDefinition(record.schema);
+            messageDeserializer = new LazyMessageReader(parsedDefinitions);
+          } else if (record.schema === "ros2") {
+            const parsedDefinitions = parseMessageDefinition(record.schema, {
+              ros2: true,
+            });
+            messageDeserializer = new ROS2MessageReader(parsedDefinitions);
+          } else if (record.encoding === "protobuf") {
+            const MsgRoot = protoParse(record.schema);
```
Comment (jtbandes) [inline packages/studio-base/src/randomAccessDataProviders/McapDataProvider.ts:85]: nit
```suggestion
            const MsgRoot = protobufjs.parse(record.schema);
```

Comment (defunctzombie) [reply packages/studio-base/src/randomAccessDataProviders/McapDataProvider.ts]: https://github.com/foxglove/studio/pull/2175/commits/9c6cacb366a6dde4634bf60f02a817c691f797b7

File: packages/studio-base/src/randomAccessDataProviders/McapDataProvider.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+            if (!isEqual(existingInfo.info, record)) {
+              throw new Error(`differing channel infos for for ${record.id}`);
+            }
+            break;
+          }
+          let messageDeserializer;
+          if (record.schema === "ros1") {
+            const parsedDefinitions = parseMessageDefinition(record.schema);
+            messageDeserializer = new LazyMessageReader(parsedDefinitions);
+          } else if (record.schema === "ros2") {
```
Comment (jtbandes) [inline packages/studio-base/src/randomAccessDataProviders/McapDataProvider.ts:79]: These two probably meant to be `record.encoding ===`?

Comment (defunctzombie) [reply packages/studio-base/src/randomAccessDataProviders/McapDataProvider.ts:79]: Thats probably right tho somehow this was working? I saw you did more testing with mcap files - did you find this needed changing?

Comment (jtbandes) [reply packages/studio-base/src/randomAccessDataProviders/McapDataProvider.ts:79]: Didn't test it, I only used a protobuf mcap file. I do think they should probably be changed

Comment (defunctzombie) [reply packages/studio-base/src/randomAccessDataProviders/McapDataProvider.ts:79]: https://github.com/foxglove/studio/pull/2254

**Pull request merged**